hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2845,6 +2863,36 @@ func reconcileControlPlaneOperatorRoleBinding(binding *rbacv1.RoleBinding, role
 	return nil
 }
 
+func reconcileControlPlaneOperatorClusterRole(clusterRole *rbacv1.ClusterRole, enableCVOManagementClusterMetricsAccess bool) error {
+	clusterRole.Rules = []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{"hypershift.openshift.io"},
+			Resources: []string{"nodepools"},
+			Verbs:     []string{"get"},
+		},
+	}
+	return nil
+}
+
+func reconcileControlPlaneOperatorClusterRoleBinding(binding *rbacv1.ClusterRoleBinding, clusterRole *rbacv1.ClusterRole, sa *corev1.ServiceAccount) error {
+	configOperatorSA := cpomanifests.ConfigOperatorServiceAccount(sa.Namespace)
+	binding.RoleRef = rbacv1.RoleRef{
+		APIGroup: "rbac.authorization.k8s.io",
+		Kind:     "ClusterRole",
+		Name:     clusterRole.Name,
+	}
+
+	binding.Subjects = []rbacv1.Subject{
+		{
+			Kind:      "ServiceAccount",
+			Namespace: configOperatorSA.Namespace,
+			Name:      configOperatorSA.Name,
+		},
+	}
+
+	return nil
+}
+
```

Comment:
i think adding this rbac will be a no go. 

The cpo has access to machines and kubevirt VMs though, which might be useful... I think there might be some other options for us as well. I'll catch you in slack to discuss this in real time.